tikv: add doc for new config `max-apply-unpersisted-log-limit` (#17857)
pingcap/docs: dynamic-config.md

| `raftstore.raft-max-size-per-msg` | The soft limit on the size of a single message packet that is allowed to be generated |
| `raftstore.raft-entry-max-size` | The hard limit on the maximum size of a single Raft log |
| `raftstore.raft-entry-cache-life-time` | The maximum remaining time allowed for the log cache in memory |

> | `raftstore.max-apply-unpersisted-log-limit` | The maximum number of committed but not persisted Raft logs that are allowed to be applied |

Reviewer: ```suggestion
| `raftstore.max-apply-unpersisted-log-limit` | The maximum number of committed but not persisted Raft logs that can be applied |
```
